lab2.js

```diff
@@ -91,47 +156,146 @@ function sayHello (sb) {
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
+    switch (this.language) {
+      case hello.klingon:
+      console.log(hello.klingon);
+      break;
+      case hello.romulan:
+      console.log(hello.romulan);
+      break;
+      case hello["federation standard"]:
+      console.log(hello["federation standard"]);
+      break;
+    }
+    switch (sb.language) {
+      case hello.klingon:
+      return hello.klingon;
+      case hello.romulan:
+      return hello.romulan;
+      case hello["federation standard"]:
+      return hello["federation standard"];
+    }
 
     //TODO: put this on the SentientBeing prototype
   }
 
+SentientBeing.prototype.sayHello = sayHello;
+
 // TODO: create three subclasses of SentientBeing, one for each
 // species above (Klingon, Human, Romulan).
 
-assert((new Human()).sayHello(new Klingon()) === "nuqneH",
-  "the klingon should hear nuqneH");
+function Klingon() {
+  this.homePlanet = "Qo\"noS";
+  this.language = hello.klingon;
+}
+function Human() {
+  this.homePlanet = "Earth";
+  this.language = hello["federation standard"];
+}
+function Romulan() {
+  this.homePlanet = "Romulus";
+  this.language = hello.romulan;
```

Comment:
I like that you're referencing the romulan property on the hello object here. If the Romulan language changed, you'd only have to update it in one place.